zrml/prediction-markets/src/migrations.rs

```diff
@@ -0,0 +1,138 @@
+pub mod _0_1_1_move_storage_to_court_and_market_commons {
+    use crate::{Config, Pallet};
+    use alloc::vec::Vec;
+    use frame_support::{
+        dispatch::Weight,
+        traits::{Get, GetPalletVersion, PalletVersion},
+        Blake2_128Concat,
+    };
+    use zeitgeist_primitives::types::{Market, PoolId};
+    use zrml_court::CourtPalletApi;
+    use zrml_market_commons::MarketCommonsPalletApi;
+
+    const PM: &[u8] = b"PredictionMarket";
+
+    pub fn migrate<T>() -> Weight
+    where
+        T: Config,
+    {
+        let mut weight: Weight = 0;
+        let version_0_1_0 = PalletVersion {
+            major: 0,
+            minor: 1,
+            patch: 0,
+        };
+        let storage_version = <Pallet<T>>::storage_version().unwrap_or(version_0_1_0);
+
+        if storage_version == version_0_1_0 {
+            for (k, v) in frame_support::migration::storage_key_iter::<
+                T::BlockNumber,
+                Vec<T::MarketId>,
+                Blake2_128Concat,
+            >(PM, b"MarketIdsPerDisputeBlock")
+            {
+                weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+                T::Court::insert_market_id_per_dispute_block(k, v);
+            }
+            for (k, v) in frame_support::migration::storage_key_iter::<
+                T::BlockNumber,
+                Vec<T::MarketId>,
+                Blake2_128Concat,
+            >(PM, b"MarketIdsPerReportBlock")
+            {
+                weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+                T::Court::insert_market_id_per_report_block(k, v);
+            }
+
+            for (k, v) in frame_support::migration::storage_key_iter::<
+                T::MarketId,
+                Option<Market<_, _>>,
+                Blake2_128Concat,
+            >(PM, b"Markets")
+            .filter_map(|(k, v)| Some((k, v?)))
+            {
+                weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+                T::MarketCommons::insert_market(k, v);
+            }
+            for (k, v) in frame_support::migration::storage_key_iter::<
+                T::MarketId,
+                Option<PoolId>,
+                Blake2_128Concat,
+            >(PM, b"MarketToSwapPool")
+            .filter_map(|(k, v)| Some((k, v?)))
+            {
+                weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+                T::MarketCommons::insert_market_pool(k, v);
+            }
+        }
```

Comment:
We have to remove the old storage from the prediction markets pallet, otherwise we have a zombie storage in the prediction market pallet that is not used but consumes disk space. I think the easiest way is to move the complete prefix inside a storage: [`move_storage_from_pallet`](https://releases.parity.io/substrate-rustdoc/frame_support/storage/migration/fn.move_storage_from_pallet.html). You can also remove the complete storage prefix, after you completed the migration (could be useful if you only want to move certain (k,v) pairs): [`remove_storage_prefix`](https://releases.parity.io/substrate-rustdoc/frame_support/storage/migration/fn.remove_storage_prefix.html)